Source/Core/DolphinWX/RmObjAddEdit.cpp

```diff
@@ -0,0 +1,359 @@
+
+	// Put the valid entry in the RmObjEntry
+	(*itCurEntry).value_upper = value_upper;
+	(*itCurEntry).value_lower = value_lower;
+
+	if (selection == -1)
+	{
+		RmObjEngine::RmObj newRmObj;
+		//newRmObj.name = WxStrToStr(EditRmObjName->GetValue());
+		newRmObj.entries = tempEntries;
+		newRmObj.active = true;
+		rmObjCodes.push_back(newRmObj);
+
+		RmObjEngine::ApplyRmObjs(rmObjCodes);
+
+		rmObjCodes.pop_back();
+	}
+	else
+	{
+		rmObjCodes.at(selection).name = WxStrToStr(EditRmObjName->GetValue());
+		rmObjCodes.at(selection).entries = tempEntries;
+
+		RmObjEngine::ApplyRmObjs(rmObjCodes);
+	}
+
+	event.Skip();
+}
+
+bool CRmObjAddEdit::UpdateTempEntryData(std::vector<RmObjEngine::RmObjEntry>::iterator iterEntry)
+{
+	unsigned long long value_lower;
+	unsigned long long value_upper;
+
+	RmObjEngine::RmObjType tempType =
+		(*iterEntry).type = (RmObjEngine::RmObjType)EditRmObjType->GetSelection();
+
+	size_t length = EditRmObjValue->GetValue().Length();
+
+	//Parse Upper If Needed
+	if (tempType > RmObjEngine::RMOBJ_64BIT) {
+		if (length > 16) {
+			if (EditRmObjValue->GetValue().Left(length - 16).ToULongLong(&value_upper, 16)) 
+			{
+				if (ParseValue(value_upper, (RmObjEngine::RmObjType)(tempType-8)))
+				{
+					(*iterEntry).value_upper = value_upper;
+					length = 16;
+				}
+				else
+				{
+					return false;
+				}
+			}
+			else 
+			{
+				wxMessageBox(_("Invalid or non hex (0-9 or A-F) character entered.\nEntry not modified."), _("Error"));
+				return false;
+			}
+		}
+		else 
+		{
+			value_upper = 0;
+		}
+	}
+	//Always Parse Lower
+	if (EditRmObjValue->GetValue().Right(length).ToULongLong(&value_lower, 16))
+	{
+		if (ParseValue(value_lower, tempType))
+		{
+			(*iterEntry).value_lower = value_lower;
+			return true;
+		}
+		else
+		{
+			return false;
+		}
+	}
+	else
+	{
+		wxMessageBox(_("Invalid or non hex (0-9 or A-F) character entered.\nEntry not modified."), _("Error"));
+		return false;
+	}
+}
+
+
+//Checks if the value is larger than the maximum it is allowed by the type.
+//e.g. if tempType = 8bits, value cannot be > 0xFF.
+bool CRmObjAddEdit::ParseValue(unsigned long long value, RmObjEngine::RmObjType tempType)
+{
+	bool parsed_ok = true;
+
+	//u64 mask = (1 << ((tempType + 1) << 3)) - 1;
+	
+	if (tempType == RmObjEngine::RMOBJ_08BIT && value > 0xff)
+		parsed_ok = false;
+	else if (tempType == RmObjEngine::RMOBJ_16BIT && value > 0xffff)
+		parsed_ok = false;
+	else if (tempType == RmObjEngine::RMOBJ_24BIT && value > 0xffffff)
+		parsed_ok = false;
+	else if (tempType == RmObjEngine::RMOBJ_32BIT && value > 0xffffffff)
+		parsed_ok = false;
+	else if (tempType == RmObjEngine::RMOBJ_40BIT && value > 0xffffffffff)
+		parsed_ok = false;
+	else if (tempType == RmObjEngine::RMOBJ_48BIT && value > 0xffffffffffff)
+		parsed_ok = false;
+	else if (tempType == RmObjEngine::RMOBJ_52BIT && value > 0xffffffffffffff)
+		parsed_ok = false;
+	else if (tempType == RmObjEngine::RMOBJ_64BIT && value > 0xffffffffffffffff)
```

Comment:
I tried to remove this duplicated code, but had problems because it is greater the 32-bits.  Whenever I tried to use other methods to do this, it would always misbehave above 32-bits, even if I stored the values in a long long or a u64.  Any suggestions to get around this would be welcome!